### PR TITLE
[analyzer] Check C++ base or member initializer in WebKit checkers.

### PR DESCRIPTION
**Pull request by rniwa**

### CODE REVIEW

File: clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.cpp [hunk 1]
```diff
@@ -525,11 +525,19 @@ bool TrivialFunctionAnalysis::isTrivialImpl(
   if (!IsNew)
     return It->second;
 
+  TrivialFunctionAnalysisVisitor V(Cache);
+
+  if (auto *CtorDecl = dyn_cast<CXXConstructorDecl>(D)) {
+    for (auto *CtorInit : CtorDecl->inits()) {
+      if (!V.Visit(CtorInit->getInit()))
```
Comment (haoNoQ) [inline clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.cpp:532]: I don't remember if either of these needs a null check. IIRC it's probably ok but if you run into crashes you know where to look ;)

**Review by haoNoQ — APPROVED** [review]
LGTM!

File: clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.cpp (hunk 1 above)
Comment (rniwa) [reply clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.cpp:532]: Doesn't seem to hit any crashes as far as building WebKit goes :)

**Comment by rniwa** [comment]
Thanks for the review!

**Pull request merged**